commonlibktx/src/androidTest/java/com/crisolutions/commonlibraryktx/ExtensionTests.kt

```diff
@@ -0,0 +1,42 @@
+package com.crisolutions.commonlibraryktx
+
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+class ExtensionTests {
+
+    @Test
+    fun testFilterIf() {
+
+        val name = "John"
+
+        name.filterIf { it == "Johns" }?.let {
+            assertTrue(it == "John")
+        } ?: fail()
+    }
+
+    @Test
+    fun testFilterIfFailure() {
+        val lastName = "Doe"
+
+        lastName
+                .map { it.toUpperCase() }
+                .joinToString()
+                .replace(",", "")
+                .replace(" ", "")
+                .filterIf { it == "DOE" }?.let {
+                    assertTrue(it == "DOE")
+                } ?: fail()
+    }
+
+    @Test
+    fun testTransform() {
+
+        val music = "Jazz"
+
+        assertBlock {
+            music.transform { it.toUpperCase() } == "JAZZ"
```

Comment:
I'd also like to see a different demonstration of how to use this since String already has a `map` extension function.

Reply:
So Im with you maybe on the confusion. Maybe I could rename this to `mapTo` or something. Basically a `map` would iterate over each character on the string, the idea behind `transform` was that you could `String -> { apply transform} -> Transforms to something else` entirely. So the toUppercase is applied to the whole string instead. Again, the idea was that it could be useful in longer chained statements. I'll think of some more examples, this could just be used as an extension function in this eg.